### PR TITLE
Updated links to SQL Server security content.

### PR DESCRIPTION
**Pull request by douglaslMS**
Removed two outdated links to old SQL Server security content and replaced them with the correct, up-to-date link to the overview page for SQL Server and Azure SQL Database security.

FYI I used a full HTTPS URL since I don't believe we can (or don't know how to) easily link cross-repo with a relative link. If that's possible, please educate me.

### CODE REVIEW

File: docs/framework/data/adonet/sql/sql-server-security.md [hunk 1]
```diff
@@ -41,12 +41,9 @@ ms.workload:
  Describes security considerations for [!INCLUDE[ssNoVersion](../../../../../includes/ssnoversion-md.md)] Express.  
   
 ## Related Sections  
- [Security and Protection (Database Engine)](http://msdn2.microsoft.com/library/bb510589\(SQL.100\).aspx.)  
- [!INCLUDE[ssNoVersion](../../../../../includes/ssnoversion-md.md)] Books Online security topics.  
-  
- [Security Considerations for SQL Server](http://go.microsoft.com/fwlink/?LinkId=98587)  
- [!INCLUDE[ssNoVersion](../../../../../includes/ssnoversion-md.md)] Books Online security topics.  
-  
+[Security Center for SQL Server Database Engine and Azure SQL Database](https://docs.microsoft.com/sql/relational-databases/security/security-center-for-sql-server-database-engine-and-azure-sql-database)
```
Comment (mairaw) [inline docs/framework/data/adonet/sql/sql-server-security.md]: In order for these links to work in the offline books you need to remove the docs.microsoft.com from the link.

So the link would look like (/sql/relational-databases/security/security-center-for-sql-server-database-engine-and-azure-sql-database)

But I need to file a feature request for a better way to doing these links

Comment (douglaslMS) [reply docs/framework/data/adonet/sql/sql-server-security.md]: @mairaw Thanks for the guidance. I've committed a new version with the fix. I don't need the ".md" file extension, do I?

**Review by mairaw — APPROVED** [review]
Thanks for making the change @douglaslMS. I've also filed the feature request now since it's a pain point for us. I'd much rather prefer we do the links the way you did it first.

**Comment by mairaw** [comment]
nits while reviewing the topic on [review](https://review.docs.microsoft.com/en-us/dotnet/framework/data/adonet/sql/overview-of-sql-server-security?branch=pr-en-us-4530): 

- should we also link to https://docs.microsoft.com/en-us/sql/sql-server/install/security-considerations-for-a-sql-server-installation @douglaslMS? 
- the last link on the See also section is not really a dev center. Remove it or update link text?

**Comment by douglaslMS** [comment]
@mairaw The built topic looks good now. What do I do about the "verify-build-before-merge" label?

**Comment by mairaw** [comment]
Our builds take a while (sometimes hours) so we add the label to remind ourselves to just check the result and merge if everything looks good. I'm doing that right now. Thanks @douglaslMS!

**Pull request merged**